client/shipping-label/views/steps/address/index.js

```diff
@@ -1,48 +1,79 @@
 import React, { PropTypes } from 'react';
-import FoldableCard from 'components/foldable-card';
 import { translate as __ } from 'lib/mixins/i18n';
 import AddressFields from './fields';
-import ExpandButton from 'shipping-label/views/expand-button';
+import { hasNonEmptyLeaves } from 'lib/utils/tree';
+import Card from 'shipping-label/views/card';
+import isEqual from 'lodash/isEqual';
+
+const renderSummary = ( { values, isValidated, normalized, pickNormalized, storeOptions, errors } ) => {
+	if ( hasNonEmptyLeaves( errors ) || ( isValidated && ! normalized ) ) {
+		return __( 'Invalid address' );
+	}
+	const { countriesData } = storeOptions;
+	const { city, postcode, state, country } = ( normalized && pickNormalized ) ? normalized : values;
+	// Summary format: "city, postcode state, country"
```

Comment:
So, ZIP at the end (I have no idea how a US address is formatted), and only put the country in the destination, **and** only if it's different of the origin country. Makes sense :)